Yoast/wordpress-seo
# P3-79 Add accessible tabular representation of the areachart data

## Context


* Provide an accessible representation of the area chart SVG.

## Summary


This PR can be summarized in the following changelog entry:

* Adds a data table to the AreaChart component to represent the chart data in an accessible way.

## Relevant technical choices:

*

## Test instructions

This PR can be tested by following these steps:

- build this branch
- click "Get related keyphrases" to open the SEMrush modal
- see the keyphrases table
- check the there are no layout problems
- check there's no horizontal scrolling
- observe each SVG area chart has now an accompanying visually hidden data table
- in `js/src/components/modals/KeyphrasesTable.js` add `isDataTableVisuallyHidden={ false }` to the AreaChart component  to make the table visible (only for testing purposes)
- build and reopen the modal
- check the area charts tables are well formed
- check the data are in a range from 0 to 100 
- regarding the **Trend** data, the SEMrush team specified "these are coefficients of the keyword volume.
1 - it is the coefficient of the highest volume for this keyword in the last 12 months, and coefficients of other months are calculated based on this maximum volume." so multiplying them by 100 made sense to me to make them a bit more meaningful
- the data table `<caption>` element gives the table a title "Keyphrase volume in the last 12 months on a scale from 0 to 100."

**Screen reader testing**
- remove `isDataTableVisuallyHidden={ false }` and build 
- quickly check with Safari and VoiceOver
  - open the modal
  - navigate to the keyphrases table by pressing Control Option Right arrow as many times as necessary
  - once within the main table, you can navigate the table cells "spatially" (in any direction) by using Control Option and the arrow keys
  - go to the trends column
  - check the SVG chart is ignored
  - observe the visually hidden table is announced
  - move with Control Option and the arrow keys through the visually hidden table and _hear_ the tabel headers and data being announced

Screenshot:

<img width="978" alt="Screenshot 2020-08-03 at 10 17 46" src="https://user-images.githubusercontent.com/1682452/89168294-f289ce80-d57c-11ea-9192-19437e3db00b.png">


## UI changes

## Documentation

## Quality assurance


Fixes P3-79
